pkg/lock/postgres/postgres.go

```diff
@@ -106,21 +118,32 @@ func (l *Lock) Unlock(s *terraform.State) (bool, error) {
 
 	defer tx.Rollback()
 
-	var lock []byte
+	var rawLock []byte
 
-	if err := tx.QueryRow(`SELECT lock_data FROM `+l.table+` WHERE state_id = $1`, s.ID).Scan(&lock); err != nil {
+	if err := tx.QueryRow(`SELECT lock_data FROM `+l.table+` WHERE state_id = $1`, s.ID).Scan(&rawLock); err != nil {
 		if err == sql.ErrNoRows {
 			return false, nil
 		}
 
 		return false, err
 	}
 
-	if string(lock) != string(s.Lock) {
+	var lock terraform.LockInfo
+
+	if err := json.Unmarshal(rawLock, &lock); err != nil {
+		return false, err
+	}
+
+	if !lock.Equal(s.Lock) {
 		return false, nil
 	}
 
-	if _, err := tx.Exec(`DELETE FROM `+l.table+` WHERE state_id = $1 AND lock_data = $2`, s.ID, s.Lock); err != nil {
+	lockBytes, err := json.Marshal(s.Lock)
```

Comment:
If lock equals s.lock, couldn't we use rawLock here instead of marshaling it again?

Reply:
Good catch!